src/backdoortrace.rs

```diff
@@ -0,0 +1,63 @@
+struct FakeBacktrace {
+    internal: FakeInternalBacktrace,
+}
+
+struct FakeInternalBacktrace {
+    backtrace: Option<MaybeResolved>,
+}
+
+struct MaybeResolved {
+    _resolved: std::sync::Mutex<bool>,
+    backtrace: std::cell::UnsafeCell<backtrace::Backtrace>,
+}
+
+pub fn backdoortrace(_opaque: &failure::Backtrace) -> Option<&backtrace::Backtrace> {
+    let _ = format!("{}", _opaque); // forces resolution
+    let no_longer_opaque: &FakeBacktrace =
+        unsafe { &*(_opaque as *const failure::Backtrace as *const FakeBacktrace) };
+    if let Some(bt) = &no_longer_opaque.internal.backtrace {
+        let bt = unsafe { &*bt.backtrace.get() };
+        return Some(bt);
+    }
+
+    None
+}
+
+pub fn print_failure_backtrace(
```

Comment:
I can haz doc comment explaining what it does as well as the risks of the current impl?

Reply:
yis